### ISSUE TITLE
Cannot log in to OSIO - seems to be limited to users in Brisbane

### ISSUE THREAD

**Issue by mishaone** [openshiftio/openshift.io#1402]
There doesn't seem to be anyone around on MM so logging an issue. I'm trying to log into OSIO this morning (worked fine last I used it on Friday) and 9 times out of ten it redirects me back to the OSIO main page. One time it let me log in. I logged back out to retry a few more times and it's back to not letting me log in. Steps:

1. Try to log in.
![login1](https://user-images.githubusercontent.com/926572/32998073-a85b98be-cde3-11e7-8703-8fd405cd4264.png)

2. Loading page appears as usual.
![login2](https://user-images.githubusercontent.com/926572/32998082-c3667174-cde3-11e7-85d2-3f2a0d8654f2.png)

3. Redirected back to the main OSIO page.
![login3](https://user-images.githubusercontent.com/926572/32998102-e2fad912-cde3-11e7-9943-2ecf7b528596.png)

I'm using this with Chrome Incognito mode on Fedora 26.

**Comment by mishaone** [comment]
For anyone else who finds this, the workaround seems to be to use https://openshift.io/openshiftio to log in and this works. The above problem still persists for me after a few more tries with fresh incognito windows.

**Comment by joshuawilson** [comment]
Could be a latency issue with everyone having the issue in Brisbane. 

@mindreeper2420 could your reset the redirects to what they were before, once logged in always go to the `_home` page? At least until we figure out what is going on.

**Comment by mindreeper2420** [comment]
@joshuawilson - best to just revert that commit/merge then.

**Comment by joshuawilson** [comment]
@mindreeper2420 your commit was not atomic, best if you make a new commit in the morning. That will also give others a change to work on the issue while it exists.

**Comment by alexeykazakov** [comment]
Adding team/platform because I don't see any issues with Auth backend. Auth team also didn't change anything in UI, so, we need to work together with the platform UI team on this issue.

**Comment by sthaha** [comment]
@sbose78 and I digged a bit into the source and here is what we think is the reason
[`document.ready`](https://github.com/openshiftio/openshift.io/blob/07ac1a837938835923361fa3a39e04f7de7884cd/src/app/index.ts#L338) calls updateUserMenu twice [1](https://github.com/openshiftio/openshift.io/blob/07ac1a837938835923361fa3a39e04f7de7884cd/src/app/index.ts#L201) [2](https://github.com/openshiftio/openshift.io/blob/07ac1a837938835923361fa3a39e04f7de7884cd/src/app/index.ts#L201)  if the method returns an error other that `401`  [user is logged out](https://github.com/openshiftio/openshift.io/blob/master/src/app/index.ts#L217) and hence if the XHR is **cancelled** for any reason the user will be logged out.

When the page is loaded[ a timer is set to redirect](https://github.com/openshiftio/openshift.io/blob/07ac1a837938835923361fa3a39e04f7de7884cd/src/app/index.ts#L175) the user to `_gettingstarted` page after a second and when this happens all XHR calls are cancelled by the browser and it results in `getUser` being cancelled if it takes more than a second (happens when the latency is high) which invokes [`this.logout`](https://github.com/openshiftio/openshift.io/blob/07ac1a837938835923361fa3a39e04f7de7884cd/src/app/index.ts#L221) and it deletes the token.

**Comment by sbose78** [comment]
yeah, we see that the request to `/api/user` gets `cancelled` because of the redirect to `GettingStarted` ,
and that might cause the control to go into the `else` block which has the `this.logout()`

```
this.getUser(this.authToken, (response: any) => {
        let user = response.data;
        // do stuff
        // do more stuff
      },
        (response: JQueryXHR, textStatus: string, errorThrown: string) => {
          if (response.status == 401) {
            this.refreshToken();
          } else {
            this.logout();
          }
        }
      );
```

**Comment by sbose78** [comment]
Here's what we need to do, to avoid the auto-logout 
https://github.com/openshiftio/openshift.io/pull/1409/files

**Comment by joshuawilson** [comment]
This has been fixed in #1413
